command/agent/auth/aws/aws.go

```diff
@@ -155,6 +155,14 @@ func NewAWSAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
 		a.lastCreds = creds
 
 		go a.pollForCreds(accessKey, secretKey, sessionToken, credentialPollIntervalSec)
+	} else {
```

Comment:
Even though it's not relevant for IAM I think this should still go up with the other config blocks instead of in an else block here.